File: Java/src/main/java/Solution901_1000/Solution976.java
package Solution901_1000;

import java.util.Arrays;

/**
 * @author windows
 */
public class Solution976 {
    public int largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        for (int i = nums.length - 1; i > 1; i--) {
            for (int j = i - 1; j > 0; j--) {
                for (int k = j - 1; k >= 0 ; k--) {
                    if (nums[j] + nums[k] > nums[i]){
                        return nums[j] + nums[k] + nums[i];
                    }
                }
            }
        }
        return 0;
    }


    public static void main(String[] args) {
        Solution976 solution976 = new Solution976();
    }
}

File: C++/src/Solution901_1000/Solution976.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution976.h"
#include <algorithm>
#include <iostream>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <vector>
using namespace std;

int largestPerimeter(vector<int>& nums) {
  sort(nums.begin(), nums.end());
  for (int i = nums.size() - 1; i >= 2; --i) {
    if (nums[i - 1] + nums[i - 2] > nums[i])
      return nums[i - 1] + nums[i - 2] + nums[i];
  }
  return 0;
}